src/corelibs/U2View/src/ov_msa/move_to_object/MoveToObjectMaController.cpp

```diff
@@ -127,4 +129,68 @@ void MoveToObjectMaController::buildMenu(GObjectView *, QMenu *menu, const QStri
 }
 
+void MoveToObjectMaController::runMoveSelectedRowsToNewFileDialog() {
+    // Get the file name to move rows to first.
+    LastUsedDirHelper lod;
+    DocumentFormatConstraints formatConstraints;
+    formatConstraints.supportedObjectTypes << GObjectTypes::MULTIPLE_SEQUENCE_ALIGNMENT;
+    QString filter = DialogUtils::prepareDocumentsFileFilter(formatConstraints, false);
+    QString selectedFilter = DialogUtils::prepareDocumentsFileFilter(BaseDocumentFormats::CLUSTAL_ALN, false);
+    lod.url = U2FileDialog::getSaveFileName(ui, tr("Select a new file to move selected rows"), lod, filter, &selectedFilter);
+    CHECK(!lod.url.isEmpty(), );
+
+    QString url = lod.url;
+    QFileInfo urlInfo(url);
+    QString fileExtension = urlInfo.suffix();
+    DocumentFormatRegistry *formatRegistry = AppContext::getDocumentFormatRegistry();
+    DocumentFormat *format = formatRegistry->selectFormatByFileExtension(fileExtension);
+    if (format == nullptr) {
+        format = formatRegistry->getFormatById(BaseDocumentFormats::CLUSTAL_ALN);
+    }
+    QStringList extensions = format->getSupportedDocumentFileExtensions();
+    if (!extensions.isEmpty() && !extensions.contains(fileExtension)) {
+        url += "." + extensions.first();
+    }
+
+    // Create a sub-alignment from moved rows.
+    QList<int> selectedViewRowIndexes = getSelection().getSelectedRowIndexes();
+    QList<int> selectedMaRowIndexes = collapseModel->getMaRowIndexesByViewRowIndexes(selectedViewRowIndexes, true);
+    QList<qint64> rowIdsToRemove = maObject->getRowIdsByRowIndexes(selectedMaRowIndexes);
+    SAFE_POINT(!rowIdsToRemove.isEmpty(), "rowIdsToRemove is empty", );
+
+    MultipleSequenceAlignment msaToExport;
+    msaToExport->setName(urlInfo.baseName());
+    msaToExport->setAlphabet(maObject->getAlphabet());
+    for (int maRowIndex : qAsConst(selectedMaRowIndexes)) {
+        const MultipleAlignmentRow &row = maObject->getRow(maRowIndex);
+        msaToExport->addRow(row->getName(), row->getSequenceWithGaps(true, true));
+    }
+
+    // Run 2 tasks: first create a new document, next remove moved rows from the original document.
+    auto createNewMsaTask = new AddDocumentAndOpenViewTask(new ExportAlignmentTask(msaToExport, url, format->getFormatId()));
+    auto removeRowsTask = new RemoveRowsFromMaObjectTask(editor, rowIdsToRemove);
+    auto task = new MultiTask(tr("Export alignment rows to a new file"), {createNewMsaTask, removeRowsTask});
+    AppContext::getTaskScheduler()->registerTopLevelTask(task);
+}
+
+/************************************************************************/
+/* RemoveRowsFromMsaObject */
+/************************************************************************/
+RemoveRowsFromMaObjectTask::RemoveRowsFromMaObjectTask(MaEditor *_maEditor, const QList<qint64> &_rowIds)
+    : Task(tr("Remove rows from alignment"), TaskFlag_RunInMainThread), maEditor(_maEditor), rowIds(_rowIds) {
```

Comment:
Why do you need to run it in the main thread? The database updating (`U2UseCommonUserModStep`) could be a bit time-consuming

Reply:
This task performs UI ops.
Is there any place in our code where U2UseCommonUserModStep are run in non-main thread?